src/main/java/org/spongepowered/asm/mixin/transformer/MixinCoprocessor.java

```diff
@@ -153,6 +155,21 @@ ProcessResult process(String className, ClassNode classNode) {
         return ProcessResult.NONE;
     }
 
+    private static final ILogger logger = MixinService.getService().getLogger("mixin");
+
+    /**
+     * Determine ahead-of-time whether a given class could be transformed ({@link ProcessResult#TRANSFORMED},
+     * {@link ProcessResult#PASSTHROUGH_TRANSFORMED}, or modification in
+     * {@link MixinCoprocessor#postProcess(String, ClassNode)}) by processing by this coprocessor.
+     * 
+     * @param className Name of the target class
+     * @return true if the coprocessor might transform the class when processed
+     */
+    public boolean processingCouldTransform(String className) {
+        logger.error("MixinCoprocessor {} does not implement processingCouldTransform, which may lead to unnecessary transformation of class {}", getName(), className);
```

Comment:
I think logging this on every classload might be slightly excessive. I had envisioned a 1-time thing